api/src/main/java/org/openmrs/api/impl/ConditionServiceImpl.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.impl;
+
+import org.openmrs.Condition;
+import org.openmrs.Patient;
+import org.openmrs.api.APIException;
+import org.openmrs.api.ConditionService;
+import org.openmrs.api.context.Context;
+import org.openmrs.api.db.ConditionDAO;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Date;
+import java.util.List;
+
+/**
+ * This class implements the {@link ConditionService} interface
+ * It defines the methods to handle the condition domain object
+ *
+ * @since 2.2
+ */
+public class ConditionServiceImpl extends BaseOpenmrsService implements ConditionService{
+
+	private ConditionDAO conditionDAO;
+
+	public void setConditionDAO(ConditionDAO conditionDAO) {
+		this.conditionDAO = conditionDAO;
+	}
+
+	public ConditionDAO getConditionDAO() {
+		return conditionDAO;
+	}
+
+	/**
+	 * Saves a condition
+	 *
+	 * @param condition - the condition to be saved
+	 */
+	@Override
+	@Transactional
+	public Condition saveCondition(Condition condition) throws APIException {
+		return conditionDAO.saveCondition(condition);
+	}
+
+	/**
+	 * Gets a condition based on the uuid
+	 *
+	 * @param uuid - uuid of the condition to be returned
+	 * @return the condition
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public Condition getConditionByUuid(String uuid) {
+		return conditionDAO.getConditionByUuid(uuid);
+	}
+
+	/**
+	 * Gets a condition by id
+	 *
+	 * @param conditionId - the id of the Condition to retrieve
+	 * @return the Condition with the given id, or null if none exists
+	 * @throws APIException
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public Condition getCondition(Integer conditionId) throws APIException {
+		return conditionDAO.getCondition(conditionId);
+	}
+
+	/**
+	 * Gets a patient's active conditions
+	 *
+	 * @param patient - the patient to retrieve conditions for
+	 * @return a list of the patient's active conditions
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public List<Condition> getActiveConditions(Patient patient) {
+		return conditionDAO.getActiveConditions(patient);
+	}
+
+	/**
+	 * Voids a condition
+	 *
+	 * @param condition  - the condition to be voided
+	 * @param voidReason - the reason for voiding the condition
+	 */
+	@Override
+	@Transactional
+	public Condition voidCondition(Condition condition, String voidReason) {
+		condition.setVoided(true);
+		condition.setVoidedBy(Context.getAuthenticatedUser());
```

Comment:
Did you look at the ObsServiceImpl's voidObs()?

Reply:
There are multiple service implementations you can take a look at for this: https://github.com/openmrs/openmrs-core/tree/master/api/src/main/java/org/openmrs/api/impl

Reply:
I did. I also considered EncounterServiceImpl's voidEncounter() method. What is your take on this @dkayiwa ?

Reply:
EncounterServiceImpl is a bit too complicated for this case

Reply:
@dkayiwa I have some reservations with the ObsServiceImpl voidObs() and unvoidObs(). This is because they both practically just save the Obs.